UnityProject/Assets/Scripts/Items/Weapons/PowerSink.cs

```diff
@@ -0,0 +1,160 @@
+using System.Collections;
+using System.Collections.Generic;
+using Systems.Explosions;
+using Items.Devices;
+using UnityEngine;
+using Communications;
+using Managers;
+using Mirror;
+using Systems.Electricity;
+using Systems.Electricity.NodeModules;
+
+namespace Items.Weapons
+{
+	public class PowerSink : SignalReceiver, ICheckedInteractable<HandApply>
+	{
+		[SerializeField] private float overchargeCap = 250000;
+		[SerializeField] private float explosionAmplifer = 0.15f;
+		[SerializeField] private float chargeAmplifer = 1f;
+		[SerializeField] private float voltageCheckTimeInSeconds = 0.2f;
+		[SerializeField] private SpriteDataSO activeSpriteSO;
+		[SerializeField] private SpriteDataSO inactiveSpriteSO;
+
+		private SpriteHandler spriteHandler;
+		private ObjectBehaviour objectBehaviour;
+		private Pickupable pickupable;
+
+		private ResistanceSourceModule RR;
+
+		[SyncVar] private bool isAnchored;
+		[SyncVar] private bool isCharging;
+		private float currentCharge;
+
+		private void Awake()
+		{
+			objectBehaviour = GetComponent<ObjectBehaviour>();
+			spriteHandler = GetComponentInChildren<SpriteHandler>();
+			pickupable = GetComponentInChildren<Pickupable>();
+			RR = GetComponent<ResistanceSourceModule>();
+		}
+
+		private void OnDisable()
+		{
+			UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, CheckForVoltage);
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			if (interaction.UsedObject == null)
+			{
+				ToggleActivity();
+				return;
+			}
+			if (interaction.UsedObject.Item().HasTrait(CommonTraits.Instance.Screwdriver))
+			{
+				if (isAnchored && isCharging == false) UnAnchor();
+				else Anchor();
+				SoundManager.PlayNetworkedAtPos(CommonSounds.Instance.screwdriver, gameObject.AssumedWorldPosServer());
+				return;
+			}
+			if (interaction.UsedObject.Item().HasTrait(CommonTraits.Instance.Multitool))
+			{
+				Chat.AddExamineMsg(interaction.Performer, $"This device's battery unit is holding {currentCharge.ToEngineering("V")}");
+				return;
+			}
+			if (interaction.UsedObject.TryGetComponent<RemoteSignaller>(out var signaller))
+			{
+				Emitter = signaller;
+				Frequency = signaller.Frequency;
+				Chat.AddExamineMsg(interaction.Performer, $"You pair the {interaction.UsedObject.ExpensiveName()} to this device.");
+			}
+		}
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			if (DefaultWillInteract.Default(interaction, side) == false) return false;
+			return true;
+		}
+
+		private void Anchor()
+		{
+			isAnchored = true;
+			pickupable.ServerSetCanPickup(false);
+			objectBehaviour.ServerSetPushable(false);
+			ElectricalManager.Instance.electricalSync.StructureChange = true;
+			Chat.AddLocalMsgToChat($"The {gameObject.ExpensiveName()} makes a clicking sound as it <b>anchors</b> to the ground", gameObject);
```

Comment:
Would be nice to have it actually make the sound as opposed to saying it makes a sound.